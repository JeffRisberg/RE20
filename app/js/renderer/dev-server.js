import { compose } from 'compose-middleware';
import webpack from 'webpack';
import webpackDevMiddleware from 'webpack-dev-middleware';
import webpackHotMiddleware from 'webpack-hot-middleware';
import { Auth, redirectToLogin } from 'utils';
import webpackConfig from '../../config/webpack.dev';
import paths from '../../config/paths';
import config from '../config';

function renderFullPage(assetsByChunkName) {
  return `
    <!doctype html>
    <html>
      <head>
        <meta charset="utf-8">
        <meta http-equiv="x-ua-compatible" content="ie=edge">
        <meta name="viewport" content="width=device-width, initial-scale=1">
        <title>${config.appName}</title>
        ${
          assetsByChunkName.main
            .filter((filename) => /\.css$/.test(filename))
            .map((filename) => `<link href="${paths.publicPath}${filename}" rel="stylesheet" />`)
        }
      </head>
      <body>
        <div id="root"></div>
        <script>
          window.__CONFIG__ = ${JSON.stringify(config).replace(/</g, '\\u003c')}
        </script>
        <script>
          // WARNING: See the following for security issues around embedding JSON in HTML:
          // http://redux.js.org/docs/recipes/ServerRendering.html#security-considerations
        </script>
        ${
          assetsByChunkName.main
            .filter((filename) => /\.js$/.test(filename))
            .map((filename) => `<script src="${paths.publicPath}${filename}"></script>`)
        }
      </body>
    </html>
  `;
}

function requestHandler(req, res, next) {
  const assetsByChunkName = res.locals.webpackStats.toJson().assetsByChunkName;
  Auth.initializeFromRequest(req);

  // not logged in  && request page is not redirect/mockLogin page
  if (!Auth.exists() && !req.path.includes('redirect')) {
    const isMock = Boolean(process.argv.slice(2).includes('mock'));
    return redirectToLogin(req, res, isMock);
  }
  return res.send(renderFullPage(assetsByChunkName));
}

function devServer() {
  const compiler = webpack(webpackConfig);

  return compose([
    webpackDevMiddleware(compiler, {
      publicPath: paths.publicPath,
      stats: 'normal',
      serverSideRender: true,
    }),
    webpackHotMiddleware(compiler),
    requestHandler,
  ]);
}

export default devServer;
