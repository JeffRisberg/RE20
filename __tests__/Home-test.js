jest
    .unmock('../app/js/pages/Home')
;

import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import TestUtils from 'react-addons-test-utils';

import Home from '../app/js/pages/Home';

describe('We can render a Home component', () => {
    it('renders correctly', () => {
        const content = 'content';
        const component = renderer.create(<Home>{content}</Home>);
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('contains content', () => {
        const home =
            TestUtils.renderIntoDocument(
                <div>
                    <Home />
                </div>
            );

        expect(ReactDOM.findDOMNode(home).textContent).toContain('RE03 Example');
    });

    it('allows a custom CSS class', () => {
        const home =
            TestUtils.renderIntoDocument(
                <div>
                    <Home className="dog" />
                </div>
            );

        // Walk through the DOM nodes and check class information

        const homeNode = ReactDOM.findDOMNode(home).children[0];
        expect(homeNode.getAttribute('class')).toEqual('dog');

        const rowNode = homeNode.children[1];
        expect(rowNode.getAttribute('class')).toEqual('row');
    });
});