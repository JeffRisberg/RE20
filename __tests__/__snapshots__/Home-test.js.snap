// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`We can render a Home component renders correctly 1`] = `
<div
  className={undefined}
>
  <h2>
    RE03 Example
  </h2>
  <div
    className="row"
  >
    <div
      className="col-md-4"
    >
      Uses React for presentation
    </div>
    <div
      className="col-md-4"
    >
      Uses Redux for data management
    </div>
    <div
      className="col-md-4"
    >
      Uses Node backend, with NEDB as proxy database
    </div>
  </div>
</div>
`;
